### PR TITLE
Add architecture overview

### PR DESCRIPTION
**Pull request by creydr**
## Proposed Changes

- :gift: Add overview over architecture of EKB

The provided document is intended to give developers a better overview over EKBs main components to get started with development.

### CODE REVIEW

File: docs/broker/README.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+The `kafka-controller` is kind of the heart of the Broker for Apache Kafkas control plane and has the following responsibilities:
+
+* Watch the following resources and update the `kafka-broker-brokers-triggers` configmap (step 3 in the diagram) with relevant information about the resource:
+  * `Broker`s with the `eventing.kantive.dev/broker.class: Kafka` annotation (step 2 in the diagram),
+  * `Trigger`s referencing a Knative Eventing Broker for Apache Kafka,
+  * `KafkaChannel`s,
+  * `KafkaSink`s 
+  * `KafkaSource`s
+* Update the status on `Broker` resources with the `eventing.kantive.dev/broker.class: Kafka` annotation with the address for the broker ingress.
```
Comment (matzew) [inline docs/broker/README.md:17]: I think I'd also list that `kafka-broker-config` there, and mention that if not provided we use the one on the SYSTEM NS.

I am a huge fan of having a 1:1 coupling of CM-Broker

Comment (matzew) [reply docs/broker/README.md:17]: I'd also argue not not use `default` NS, but a different one, to make things a bit more explicit?

File: docs/broker/README.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
+# Knative Eventing Broker for Apache Kafka
+
+This document gives a rough overview over the architecture of the Knative Eventing Broker for Apache Kafka.
+
+![Overview](eventing-kafka-broker.png)
```
Comment (matzew) [inline docs/broker/README.md:5]: I do not see `(7)` in the diagram, just `(6)` and `(8)`  :upside_down_face:

File: docs/broker/README.md (hunk 1 above)
Comment (creydr) [reply docs/broker/README.md:17]: > I'd also argue not not use default NS, but a different one, to make things a bit more explicit?

:+1: updated

**Comment by creydr** [comment]
Hi @matzew @pierDipi @aliok,
could you take a look at this PR and give me your opinions on it?

/assign @matzew @pierDipi @aliok

**Review by matzew — APPROVED** [review]
/lgtm
/approve

/retest

I like it very much! Thanks a lot for this

**Comment by pierDipi** [comment]
Thanks folks, this is great!

**Pull request merged**